celery/backends/redis.py

```diff
@@ -361,7 +364,7 @@ def on_chord_part_return(self, request, state, result,
                 decode, unpack = self.decode, self._unpack_chord_result
                 with client.pipeline() as pipe:
                     resl, = pipe \
-                        .lrange(jkey, 0, total) \
+                        .zrangebyscore(jkey, '-inf', '+inf') \
```

Comment:
You could also use:

```python
zrange(jkey, 0, -1)
```

I think it would also make the mock client https://github.com/celery/celery/pull/4858/files#diff-5cb790f5c0c0aea16658640ccd102c19R97 a bit simpler to implement. Not really important, merely a suggestion though.